### PR TITLE
core/ts_version_issue_422_create_order_endpoint_test

### PR DESCRIPTION
**Pull request by GreatGaryCooper**
## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [x] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: ts-version/src/test/e2e/create_order.e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+    const res = await axios.post(
+      `http://localhost:${serverPort}/user/order`,
+      createOrderPayload,
+      {
+        headers: { Authorization: `Bearer ${jwtToken}` },
+      },
+    );
+
+    expect(res.data.network).toEqual(createOrderPayload.network);
+    expect(res.data.id).toEqual(1);
```
Comment (WilliamHawkingStephen) [inline ts-version/src/test/e2e/create_order.e2e.test.ts:675]: every thing looks great nice job :+1: 

suggestion: 
maybe we can use toBe for primitives like strings, numbers or booleans

**Comment by WilliamHawkingStephen** [comment]
we have a loop detected error :
`if (matchError) throw new LoopDetectedError("Order is starved");`
which is for starve orders.
i think maybe we need to test this one

**Comment by WilliamHawkingStephen** [comment]
suggestion: 
maybe we can add test case about Undefined order id when a order id not found in match function: 
`if (!order.id) throw new BadRequestError("Undefined order id");` 
order adapter => line 295

**Comment by GreatGaryCooper** [comment]
> we have a loop detected error : `if (matchError) throw new LoopDetectedError("Order is starved");` which is for starve orders. i think maybe we need to test this one

You're right @WilliamHawkingStephen , but I think this should be moved to the `getFills` endpoint since in order to face that error, there should be failed fills I guess...

Will investigate more, thanks for that 🙏🏻 🌹

**Comment by GreatGaryCooper** [comment]
> suggestion: maybe we can add test case about Undefined order id when a order id not found in match function: `if (!order.id) throw new BadRequestError("Undefined order id");` order adapter => line 295

Sure, will add it.
Thanks a lot 🙏🏻

**Comment by GreatGaryCooper** [comment]
@WilliamHawkingStephen Regarding the `Undefined order id` error, the implementation needs some work, the thing is that we actually do face an error, but it's not very clear what the issue is, I created a new card for it, will add that test case, proper error handling, and request validation in that card.

Link to the card: https://github.com/orgs/dexpresso-protocol/projects/1/views/1?filterQuery=great&pane=issue&itemId=28410568

File: ts-version/src/test/e2e/create_order.e2e.test.ts (hunk 1 above)
Comment (GreatGaryCooper) [reply ts-version/src/test/e2e/create_order.e2e.test.ts:675]: Thanks for the complements buddy 🌹 

About using toBe for primitive types like strings, in the mentioned line, beside the type of the returned response, we actually do care about the value of it, for example here, we're making sure that the id of the new order is `1`.
Anyways, there is no blocking for using `toBe` method that jest provides.

Comment (WilliamHawkingStephen) [reply ts-version/src/test/e2e/create_order.e2e.test.ts:675]: > @WilliamHawkingStephen Regarding the `Undefined order id` error, the implementation needs some work, the thing is that we actually do face an error, but it's not very clear what the issue is, I created a new card for it, will add that test case, proper error handling, and request validation in that card.
> 
> Link to the card: https://github.com/orgs/dexpresso-protocol/projects/1/views/1?filterQuery=great&pane=issue&itemId=28410568

nice ! :+1:

**Comment by WilliamHawkingStephen** [comment]
> @WilliamHawkingStephen Regarding the `Undefined order id` error, the implementation needs some work, the thing is that we actually do face an error, but it's not very clear what the issue is, I created a new card for it, will add that test case, proper error handling, and request validation in that card.
> 
nice! :+1:

**Review by WilliamHawkingStephen — APPROVED** [review]

**Review by isaac-fibonacci — APPROVED** [review]

**Pull request merged**